quantumcat/gates/custom_gates/cirq/c3x_gate.py

```diff
@@ -0,0 +1,22 @@
+import cirq
+import numpy
+
+
+class C3XGate(cirq.Gate):
+
+    def __init__(self):
+        super(C3XGate, self).__init__()
+
+    def _num_qubits_(self):
+        return 4
+
+    def _unitary_(self, dtype=None):
+        mat = _compute_control_matrix(self.base_gate.to_matrix(),
+                                      self.num_ctrl_qubits,
+                                      ctrl_state=self.ctrl_state)
+        if dtype:
+            return numpy.asarray(mat, dtype=dtype)
+        return mat
+    
+    def _circuit_diagram_info_(self, args):
+        return ["C3XGate"] * self.num_qubits()
```

Comment:
Instead of [C3Xgate]*num_qubits(), change it to [C3X_c1, C3X_c2, C3X_c3, C3X_t] . Do the same for all multi qubit gates where you name each qubit as control and target